scripts/dotnet-cli-build/CompileTargets.cs

```diff
@@ -203,6 +207,25 @@ private static BuildTargetResult CompileStage(BuildTargetContext c, DotNetCli do
             File.Delete(Path.Combine(binDir, $"dotnet{Constants.ExeSuffix}"));
             File.Copy(Path.Combine(binDir, $"corehost{Constants.ExeSuffix}"), Path.Combine(binDir, $"dotnet{Constants.ExeSuffix}"));
 
+            // HACK
+            // bootstrapping for Windows x86. Copy csc/vbc from stage0.
+            // This is a temporary hack for https://github.com/dotnet/roslyn/issues/8951
+            if (IsWinx86)
```

Comment:
All of these If(Platform) or If(Architecture) spread through this code are worrisome. You don't need to deal with this now, but we need to think about how to detangle this, otherwise as we add more platforms here this will turn into a nightmare.